### ISSUE TITLE
cannot include setup.h

### ISSUE THREAD

**Issue by jfoegen** [arturasn/mdb2sqlite#1]
I just pulled down the project and just tried to build under VS 2013 and it is complaining about not finding "wx/setup.h' in the platform.h when compiling.  Right above it gets the compiler.h, but there is no setup.h in the directory directly, but looks like there are ones in each of the platforms.  Figure there is simple I am missing.

Thanks

**Comment by arturasn** [comment]
Did you build wxwidgets?

**Comment by jfoegen** [comment]
Yes, I was in error and didn't realize that needed to be build separately.  I then tried Debug DLL and was still getting some include errors even though I build both debug types in the wxwidgets solution, but I was able to build and run debug without any issues,

Thanks for the help.

**Comment by zhengzilu** [comment]
So I need to build both \bin\mdb2sqlite.sln and \mdb2sqlite-master\wxwidgets\build\msw\wx_vc12.sln in Debug DLL mode?
Both of them failed.
(I just want to create a dll version of this application so I can convert file in my program.)

**Comment by zhengzilu** [comment]
I built \wxwidgets\build\msw\wx_vc12.sln under VS 2017. Then it missed \lib\vc_lib.obj when building  \bin\mdb2sqlite.sln.
Is this a vc++ toolset version problem? Can you add a vc++2017 version 15 build file?
(I just want to create a dll version of this application to convert db in my program. ) Thanks.

**Comment by arturasn** [comment]
> I built \wxwidgets\build\msw\wx_vc12.sln under VS 2017. Then it missed \lib\vc_lib.obj when building \bin\mdb2sqlite.sln.
> Is this a vc++ toolset version problem? Can you add a vc++2017 v14.16 build file?
> (I just want to create a dll version of this application to convert db in my program. ) Thanks.

Project was using a very old version of wxWidgets, that was the main reason of your problem.
As of today mdb2sqlite project is built using CMake and wxWidgets version was updated to achieve all  current Visual Studio version support. 
You can build mdb2sqlite for vc++ 2017 by firstly building wxWidgets with that toolset, which is wxwidgets\build\msw\wx_vc15.sln and then generating mdb2sqlite project using vc++ 2017 generator which is part of the CMake infrastructure. To create a dll version of the project you will have to modify the appropriate contents of the CMakeLists.txt file to achieve this, Best of luck.

**Comment by zhengzilu** [comment]
Thank you very much. I finally managed to compile your project in VS 2013 and 2012.(but failed in VS 2010, the reason I don't think it supports C++11 auto)
And I manged to create a dll version which can be called externally to use the function bool CSettingsReader::Control(const char *Path, const char *dPath, CUIObs *pObs, bool bConvert) 
;-) But I just mask CUIObs *pObs by NULL, and comment some statements. It still rests a lot of optimisation for me to do to get rid of  wxwidegets.
Regards.
